src/Jackett.Common/Indexers/HorribleSubs.cs

```diff
@@ -145,182 +139,130 @@ private async Task<IEnumerable<ReleaseInfo>> PerformLatestQuery(TorznabQuery que
-                        var showAPIdom = ResultParser.ParseDocument(showAPIResponse.Content);
-                        var releaseRowResults = showAPIdom.QuerySelectorAll("div.rls-info-container");
-                        releaserows.AddRange(releaseRowResults);
+                        var showApiResponse = await RequestStringWithCookiesAndRetry(apiUrl + "&nextid=" + nextId, string.Empty);
+                        var showApiDom = parser.ParseDocument(showApiResponse.Content);
+                        var releaseRowResults = showApiDom.QuerySelectorAll("div.rls-info-container");
+                        rows.AddRange(releaseRowResults);
                         nextId++;
 
                         if (releaseRowResults.Length == 0 || latestOnly)
-                        {
                             break;
-                        }
                     }
                 }
 
-                foreach (var releaserow in releaserows)
+                foreach (var row in rows)
                 {
-                    var dateStr = releaserow.QuerySelector(".rls-date").TextContent.Trim();
-                    var title = releaserow.FirstChild.TextContent;
+                    var dateStr = row.QuerySelector(".rls-date").TextContent.Trim();
+
+                    var qTitle = row.QuerySelector("a");
+                    var title = qTitle.TextContent;
                     title = title.Replace("SD720p1080p", "");
                     title = title.Replace(dateStr, "");
 
                     if (!string.IsNullOrWhiteSpace(titleContains) && !title.Contains(titleContains))
-                    {
                         continue;
-                    }
 
                     // Ensure fansub group name is present in the title
                     // This is needed for things like configuring tag restrictions in Sonarr
                     if (!title.Contains("[HorribleSubs]"))
-                    {
                         title = "[HorribleSubs] " + title;
-                    }
 
-                    DateTime releasedate;
-                    if (dateStr == "Today")
-                    {
-                        releasedate = DateTime.Today;
-                    }
-                    else if (dateStr == "Yesterday")
-                    {
-                        releasedate = DateTime.Today.AddDays(-1);
-                    }
-                    else
-                    {
-                        releasedate = DateTime.SpecifyKind(DateTime.ParseExact(dateStr, "MM/dd/yy", CultureInfo.InvariantCulture), DateTimeKind.Utc).ToLocalTime();
-                    }
-
-                    var p480 = releaserow.QuerySelector(".link-480p");
+                    var episode = qTitle.QuerySelector("strong")?.TextContent;
+                    var comments = new Uri(resultUrl + (episode != null ? "#" + episode : ""));
 
-                    if (p480 != null)
+                    var publishDate = dateStr switch
```

Comment:
Would `DateTimeUtil.FromUknown()` be a better option here?

Reply:
nice catch